projects/vdk-plugins/vdk-meta-jobs/src/vdk/plugin/meta_jobs/remote_data_job.py

```diff
@@ -0,0 +1,214 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import logging
+import os
+import time
+import uuid
+from enum import Enum
+from typing import Optional
+
+from taurus_datajob_api import ApiClient
+from taurus_datajob_api import Configuration
+from taurus_datajob_api import DataJobExecution
+from taurus_datajob_api import DataJobExecutionRequest
+from taurus_datajob_api import DataJobsExecutionApi
+from urllib3 import Retry
+from vdk.plugin.control_api_auth.authentication import Authentication
+
+log = logging.getLogger(__name__)
+
+
+class JobStatus(str, Enum):
+    """
+    Enum for the possible statuses a job execution can have.
+    """
+
+    SUBMITTED = "submitted"
+    RUNNING = "running"
+    SUCCEEDED = "succeeded"
+    CANCELLED = "cancelled"
+    SKIPPED = "skipped"
+    USER_ERROR = "user_error"
+    PLATFORM_ERROR = "platform_error"
+
+
+class RemoteDataJob:
+    """
+    Interact with Verstile Data Kit (VDK) Control Service
+    """
+
+    def __init__(
+        self,
+        job_name: str,
+        team_name: str,
+        rest_api_url: str,
+        timeout: int = 5,  # TODO: Set reasonable default
+        **kwargs,
+    ) -> None:
+        self.job_name = job_name
+        self.team_name = team_name
+        self._rest_api_url = rest_api_url
+        self.timeout = timeout
+        self.deployment_id = "production"  # currently multiple deployments are not supported so this remains hardcoded
+        self.auth: Optional[Authentication] = kwargs.pop("auth", None)
+
+        # setting these manually to avoid using VDKConfig
+        # TODO: set op ID from current job
+        self.op_id = os.environ.get("VDK_OP_ID_OVERRIDE", f"{uuid.uuid4().hex}"[:16])
+        self.http_verify_ssl = os.getenv(
+            "VDK_CONTROL_HTTP_VERIFY_SSL", "True"
+        ).lower() in ("true", "1", "t")
+        self.http_connection_pool_maxsize = int(
+            os.getenv("VDK_CONTROL_HTTP_CONNECTION_POOL_MAXSIZE", "2")
+        )
+        self.http_total_retries = int(os.getenv("VDK_CONTROL_HTTP_TOTAL_RETRIES", "10"))
+        self.http_connect_retries = int(
+            os.getenv("VDK_CONTROL_HTTP_CONNECT_RETRIES", "6")
+        )
+        self.http_read_retries = int(os.getenv("VDK_CONTROL_HTTP_READ_RETRIES", "6"))
+
+        self.__execution_api = self._get_execution_api()
+
+    def start_job_execution(self, **request_kwargs) -> str:
+        """
+        Triggers a manual Datajob execution.
+
+        :param: request_kwargs: Request arguments to be included with the HTTP request
+        """
+        execution_request = DataJobExecutionRequest(
```

Comment:
Do we need a new job execution type, e.g. Meta-job-triggered, apart from the existing Scheduled and Manual? I believe we do.

Reply:
Not sure. started_by indicates it was started by meta job.
It looks like this "meta-job/aivanov" 

 I will create a separate issue for us to ponder this. But we do not have separate execute type if job is tarted from UI vs CLI currently. Meta Job is just another agent

Reply:
Created https://github.com/vmware/versatile-data-kit/issues/1272 for this